File: frontend/src/app/core/models/property.ts
export interface PropertyType {
  id: number;
  code: string;
  nom: string;
}export interface PropertyImage {

  id: number;

  propertyId: number;

  imageUrl: string;

  titre: string | null;

  principale: boolean;

  ordreAffichage: number;
}


export interface Property {

  id: number;

  reference: string;

  typeId: number;

  typeCode: string;

  typeNom: string;

  titre: string;

  description: string;

  transactionType: string;

  prix: number;

  devise: string;

  adresse?: string | null;

  quartier?: string | null;

  ville: string;

  departement: string;

  pays: string;

  latitude?: number | null;

  longitude?: number | null;

  chambres: number;

  sallesBain: number;

  superficie: number;

  statut: string;

  featured: boolean;

  mainImageUrl: string | null;

  images: PropertyImage[];

  createdAt?: string;

  updatedAt?: string;
}

File: backend/src/main/java/com/novaimmo/demo/property/Property.java
package com.novaimmo.demo.property;


import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

    @Entity
    @Table(name = "properties")
    @Getter
    @Setter
    public class Property {

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;


        @Column(nullable = false, unique = true)
        private String reference;


        @ManyToOne
        @JoinColumn(
                name = "type_id",
                nullable = false
        )
        private PropertyType type;


        @Column(nullable = false)
        private String titre;


        @Column(columnDefinition = "TEXT")
        private String description;


        @Column(name = "transaction_type")
        private String transactionType;


        private BigDecimal prix;


        private String devise;


        private String adresse;

        private String quartier;

        private String ville;

        private String departement;

        private String pays;


        private BigDecimal latitude;

        private BigDecimal longitude;


        private Integer chambres;


        @Column(name = "salles_bain")
        private Integer sallesBain;


        private BigDecimal superficie;


        private String statut;


        private Boolean featured;


        @Column(name = "created_at")
        private LocalDateTime createdAt;


        @Column(name = "updated_at")
        private LocalDateTime updatedAt;

}
